code/workspaces/web-app/src/components/settings/AddUserPermissions.js

```diff
@@ -0,0 +1,177 @@
+      <PermissionsSelector
+        className={classes.permissionsSelector}
+        permissionLevels={permissionLevels}
+        selectedPermissions={selectedPermissions}
+        setSelectedPermissions={setSelectedPermissions} />
+      <AddUserButton
+        userInformation={users.value}
+        selectedUserName={selectedUserName}
+        project="project"
+        onClickFn={dispatchAddUserAction}
+        dispatch={dispatch}
+        selectedPermissions={selectedPermissions} />
+    </div>
+  );
+}
+
+export function UsersAutofill({ userNames, setSelectedUserName }) {
+  const classes = useStyles();
+  return (
+    <Downshift>
+      {
+        ({
+          isOpen,
+          inputValue,
+          getInputProps,
+          getMenuProps,
+          getItemProps,
+        }) => {
+          setSelectedUserName(inputValue);
+          return (
+            <div className={classes.usersAutofill}>
+              <TextField
+                label="Add User"
+                placeholder="Type user's email..."
+                variant="outlined"
+                fullWidth
+                {...getInputProps()} />
+              <UsersDropdown
+                userNames={userNames}
+                inputValue={inputValue}
+                getMenuProps={getMenuProps}
+                getItemProps={getItemProps}
+                isOpen={isOpen} />
+            </div>
+          );
+        }
+      }
+    </Downshift>
+  );
+}
+
+export function UsersDropdown({ userNames, inputValue, getMenuProps, getItemProps, isOpen }) {
+  const classes = useStyles();
+  const namesMatchingSearch = userNames.filter(name => name.search(inputValue) !== -1);
+
+  const show = isOpen && namesMatchingSearch.length > 0;
+  return (
+    <Paper
+      className={show ? classes.dropDownOpen : classes.dropDownClosed}
+      {...getMenuProps()}
+    >
+      {show
+        ? namesMatchingSearch
+          .map(userName => <MenuItem {...getItemProps({ item: userName, key: userName })}>{userName}</MenuItem>)
+        : null}
+    </Paper>
+  );
+}
+
+export function PermissionsSelector({ permissionLevels, selectedPermissions, setSelectedPermissions }) {
+  const classes = useStyles();
+  return (
+    <TextField
+      className={classes.permissionsSelector}
+      label="Permissions"
+      variant="outlined"
+      select
+      value={selectedPermissions}
+      onChange={event => setSelectedPermissions(event.target.value)}
+    >
+      {permissionLevels.map(item => <MenuItem value={item} key={item}>{item}</MenuItem>)}
+    </TextField>
+  );
+}
+
+export function AddUserButton({ userInformation, selectedUserName, project, selectedPermissions, onClickFn, dispatch }) {
+  const selectedUser = userInformation.find(user => user.name === selectedUserName);
+  return (
+    <Button
+      variant="contained"
+      color="primary"
+      disabled={!selectedUser}
+      onClick={() => onClickFn(project, selectedUser, selectedPermissions, dispatch)}
+    >Add</Button>
```

Comment:
Comment on style. Interesting that you put the closing `>` here rather than on 164. From HTML days I'd expect
```
<Button>
  Add
</Button>
```

Reply:
If I put the closing tag on line 164 then I see it ending up as being 
```
<Button
  variant="outlined"
  color="primary"
  disabled={!selectedUser}
  onClick={() => onClickFn(project, selectedUser, selectedPermissions, dispatch)}>
  Add
</Button>
```
and in that I lose the fact that `Add` comes after the closing the of the `Button` tag. I find it a bit harder to scan than how I have it above!